File: Algorithms/p343_Integer_Break/p343_Integer_Break_2.cpp
class Solution {
public:
    int integerBreak(int n) {
        if(n==2){
            return 1;
        }else if(n==3){
            return 2;
        }
        int res = 1;
        for(int i=0;i<n/3-1;i++){
            res *= 3;
        }
        if(n%3==0){
            return res*3;
        }else if(n%3==1){
            return res*4;
        }else{
            return res*6;
        }
    }
};

File: Algorithms/p343_Integer_Break/p343_Integer_Break_2.py
class Solution(object):
    def integerBreak(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n==2:
            return 1
        elif n==3:
            return 2
        if n%3 == 0:
            return 3**(n//3)
        elif n%3 == 1:
            return 3**(n//3-1)*4
        else:
            return 3**(n//3)*2
